usaspending_api/search/v2/views/spending_by_category_views/spending_by_category.py

```diff
@@ -66,6 +67,18 @@ def post(self, request: Request) -> Response:
         original_filters = request.data.get("filters")
         validated_payload = TinyShield(models).block(request.data)
 
+        # Custom logic to support Subawards search for DEFC
+        # Check if it's the Spending by DEFC endpoint
```

Comment:
This logic applies to both Subawards and Transactions. See the follow following new language. 

Because Subawards and Transactions are associated with DEFCs via their Prime Award, which contains a list of all DEFCs from its associated File C records, we need to limit results to records with a `sub_action_date` and `action_date`, respectively, earlier than the enactment date of the public law for each DEFC. This provides a more accurate breakdown of funds by DEFC. This can be accomplished by simply adding an extra filter for all DEFCs because DEFC filtering logic already takes this into account in `query_with_filters.py`. Because we are grouping by DEFC, it is safe to filter down to only records that include one.